src/server/search/search_family_test.cc

```diff
@@ -361,6 +361,19 @@ TEST_F(SearchFamilyTest, Unicode) {
               UnorderedElementsAre("visits", "100", "title", "πανίσχυρη ΛΙΒΕΛΛΟΎΛΗ Δίας"));
 }
 
+TEST_F(SearchFamilyTest, UnicodeWords) {
+  EXPECT_EQ(Run({"ft.create", "i1", "schema", "title", "text"}), "OK");
+
+  Run({"hset", "d:1", "title",
+       "WORD!!! Одно слово? Zwei Wörter. Comma before ,sentence, "
+       "Τρεις λέξεις: χελώνα-σκύλου-γάτας",
+       "visits", "400"});
+
+  // Make sure it includes ALL those words
+  EXPECT_THAT(Run({"ft.search", "i1", "word слово wörter sentence λέξεις γάτας"}),
+              AreDocIds("d:1"));
```

Comment:
does it work with hebrew? :)

we do not have upper-case though.

`מה קורה קורה, אחי?`

Reply:
It does, I have a test above for unicode. This is just for word splitting, but added here one as well